components/brave_webtorrent/extension/background/reducers/webtorrent_reducer.ts

```diff
@@ -55,25 +57,40 @@ const tabUpdated = (tabId: number, url: string, state: TorrentsState) => {
   const { torrentStateMap, torrentObjMap } = state
   const origTorrentState: TorrentState = torrentStateMap[tabId]
   const origInfoHash = origTorrentState ? origTorrentState.infoHash : undefined
-  let newTorrentState
-  let newInfoHash
-
-  // delete old torrent state
-  delete torrentStateMap[tabId] // delete old torrent state
+  let newTorrentState: TorrentState | undefined
+  let newInfoHash: string | undefined
 
   // create new torrent state
   const parsedURL = new window.URL(url)
+  const torrentId = parsedURL.href
   if (parsedURL.protocol === 'magnet:') { // parse torrent
-    const torrentId = parsedURL.href
     try {
       const { name, infoHash, ix } = ParseTorrent(torrentId)
       newInfoHash = infoHash
       newTorrentState = { tabId, torrentId, name, infoHash, ix }
     } catch (error) {
       newTorrentState = { tabId, torrentId, errorMsg: error.message }
     }
+  } else if (parsedURL.protocol === 'https:' || parsedURL.protocol === 'http:') {
+    const name = parsedURL.pathname.substr(parsedURL.pathname.lastIndexOf('/') + 1)
+    let ix: number | undefined = Number(parse(parsedURL.hash).ix)
```

Comment:
nit: please reference in a comment that this is the file index to select a specific file from that torrent so other people not familiar w/ this code can understand faster what this does